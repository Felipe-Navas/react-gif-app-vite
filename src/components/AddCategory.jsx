import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (e) => {
    e.preventDefault()

    const newCategory = inputValue.trim()
    if (newCategory.length <= 2) return

    // setCategories((categories) => [inputValue, ...categories])
    onNewCategory(newCategory)
    setInputValue('')
  }

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Find gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}
