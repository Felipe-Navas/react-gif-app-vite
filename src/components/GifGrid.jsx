import PropTypes from 'prop-types'
import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {
  const { images, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {loading && <h2>Loading...</h2>}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
