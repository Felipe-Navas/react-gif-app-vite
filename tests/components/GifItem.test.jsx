import { render } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Testing GifItem', () => {
  const title = 'Test title'
  const url = 'https://media1.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif'

  test('should match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot()
  })
})
