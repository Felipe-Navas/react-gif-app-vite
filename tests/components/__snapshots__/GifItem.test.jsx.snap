// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="https://media1.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
